coffee-rest/src/main/java/hu/icellmobilsoft/coffee/rest/exception/DefaultGeneralExceptionMapper.java

```diff
@@ -82,80 +86,108 @@
     public Response toResponse(Exception e) {
         Response result = null;
         if (e instanceof BaseExceptionWrapper<?>) {
-            BaseExceptionWrapper<?> wrappedException = (BaseExceptionWrapper<?>) e;
-            if (wrappedException.getException() != null) {
-                log.info("Wrapped exception. Trying to match the correct mapper...");
-                result = handleWrappedException(wrappedException.getException());
-            } else if (e.getCause() instanceof BaseException) {
-                log.info("Wrapped BaseException. Trying to match the correct mapper...");
-                result = handleWrappedException((BaseException) e.getCause());
-            } else {
-                log.error("Unknown error in cause: ", e);
-                log.writeLogToError();
+            Exception unwrappedException = unwrapException((Exception & BaseExceptionWrapper<?>) e);
+            if (unwrappedException instanceof BaseException) {
+                result = handleWrappedException((BaseException) unwrappedException);
             }
-        } else {
-            log.error("Unknown error: ", e);
-            log.writeLogToError();
         }
         return (result != null) ? result : handleException(e);
     }
 
-    /**
-     * <p>handleException.</p>
-     */
-    protected Response handleException(Exception e) {
-
-        TechnicalFault dto = new TechnicalFault();
-        exceptionMessageTranslator.addCommonInfo(dto, e, CoffeeFaultType.OPERATION_FAILED);
-
-        Response.Status statusCode = Response.Status.INTERNAL_SERVER_ERROR;
-        if (e instanceof InternalServerErrorException) {
-            statusCode = Response.Status.BAD_REQUEST;
-        }
-        ResponseBuilder responseBuilder = Response.status(statusCode);
-        boolean productionStage = ProjectStage.Production.equals(projectStage);
-        if (productionStage) {
-            handleProductionStageException(dto, responseBuilder, e);
+    protected <WRAPPED extends Exception & BaseExceptionWrapper<?>> Exception unwrapException(WRAPPED wrappedException) {
+        Exception unwrapped;
+        if (wrappedException.getException() != null) {
+            log.info("Wrapped exception.");
```

Comment:
ezt info szinten ki akarjuk iratni minden alkalommal? látom h korábban is benne volt, de lehet jobb lenne trace szinten kiloggolni